File: tps/0/bench/js/find.js
import { measureExecutionTime, getArray } from "./bench.js";
import { BENCH_ARGS } from "./args.js";

measureExecutionTime(
	"array#find",
	getArray,
	(array) => array.find((value) => value == 569),
	BENCH_ARGS.WM_ITERATIONS,
	BENCH_ARGS.ITERATIONS
);

File: tps/0/bench/haskell/app/Find.hs
module Main where
import Data.List(find)
import Bench (bench, list) 
import Args (getCLIArgs)

findEl :: Int -> Maybe Int
findEl el = find (==el) list

main :: IO ()
main = do
  [wmIterations, iterations] <- getCLIArgs
  bench "list#find" findEl 569 wmIterations iterations
 
